nixos/modules/services/networking/adguardhome.nix

```diff
@@ -0,0 +1,88 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.adguardhome;
+
+  args = concatStringsSep " " ([
+    "--no-check-update"
+    "--pidfile /run/AdGuardHome/AdGuardHome.pid"
+    "--work-dir /var/lib/${cfg.stateDir}"
+    "--config /var/lib/${cfg.stateDir}/AdGuardHome.yaml"
+    "--host ${cfg.host}"
+    "--port ${toString cfg.port}"
+  ] ++ cfg.extraArgs);
+
+in
+{
+  options.services.adguardhome = with types; {
+    enable = mkEnableOption "AdGuard Home network-wide ad blocker";
+
+    stateDir = mkOption {
+      default = "AdGuardHome";
+      type = str;
+      description = ''
+        Directory below <literal>/var/lib</literal> to store AdGuard Home's data
+        and configuration."
+      '';
+    };
```

Comment:
If it's below `/var/lib` anyway, there's no use in making this configurable, is there?

Reply:
I strongly agree with @dotlambda on this.

Reply:
I'll get rid of it 🗑️